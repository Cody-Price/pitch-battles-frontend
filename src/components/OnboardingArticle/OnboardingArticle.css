.onboarding-article {
  margin-top: 2rem;
  width: 600px;
}

.onboarding-image {
  height: 200px;
  width: 200px;
  box-shadow: 1px 1px 1px #4776bb;
}

.onboarding-content {
  margin: 1rem 2rem;
}

@media (max-width: 700px) {
  .onboarding-article {
    margin-top: 4rem;
    width: 340px;
  }

  .onboarding-image {
    height: 150px;
    width: 150px;
    box-shadow: 1px 1px 1px #4776bb;
  }
}
