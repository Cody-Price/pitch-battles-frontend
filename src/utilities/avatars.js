const avatars = [
  "one",
  "one",
  "two",
  "three",
  "four",
  "five",
  "six",
  "seven",
  "eight",
  "nine",
  "ten",
  "eleven",
  "twelve",
  "thirteen",
  "fourteen"
];

export default avatars;
