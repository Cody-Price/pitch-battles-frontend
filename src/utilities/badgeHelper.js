export const badgeHelper = [
  { name: "level_one", badge: "level one: completed", rank: 2 },
  { name: "level_one_perfect", badge: "level one: perfect", rank: 3 },
  { name: "level_two", badge: "level two: completed", rank: 4 },
  { name: "level_two_perfect", badge: "level two: perfect", rank: 6 },
  { name: "level_three", badge: "level three: completed", rank: 7 },
  { name: "level_three_perfect", badge: "level three: perfect", rank: 9 },
  { name: "level_four", badge: "level four: completed", rank: 11 },
  { name: "level_four_perfect", badge: "level four: perfect", rank: 12 },
  { name: "all_perfect", badge: "all levels: perfect", rank: 14 },
  { name: "five_games", badge: "play 5 games", rank: 1 },
  { name: "ten_games", badge: "play 10 games", rank: 5 },
  { name: "twenty_games", badge: "play 20 games", rank: 8 },
  { name: "fifty_games", badge: "play 50 games", rank: 10 },
  { name: "one_hundred_games", badge: "play 100 games", rank: 13 },
  { name: "two_hundred_games", badge: "play 200 games", rank: 15 },
  { name: "five_hundred_games", badge: "play 500 games", rank: 16 },
  { name: "one_thousand_games", badge: "play 1000 games", rank: 17 }
];
