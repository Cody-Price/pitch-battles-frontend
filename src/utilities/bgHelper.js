export const bgHelper = [
  "one",
  "two",
  "three",
  "four",
  "five",
  "six",
  "seven",
  "eight"
];
