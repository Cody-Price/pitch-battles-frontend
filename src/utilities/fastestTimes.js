export const fastestTimes = user => {
  console.log(user);
};
