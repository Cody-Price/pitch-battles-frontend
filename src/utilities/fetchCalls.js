export const login = async body => {
  const url = "https://pitch-battles-api.herokuapp.com/login";

  const options = {
    body: JSON.stringify(body),
    method: "POST",
    mode: "cors",
    headers: {
      "Content-Type": "application/json"
    }
  };

  const response = await fetch(url, options);
  return await response.json();
};

export const signUp = async body => {
  const url = "https://pitch-battles-api.herokuapp.com/api/v1/users";

  const options = {
    body: JSON.stringify(body),
    method: "POST",
    mode: "cors",
    headers: {
      "Content-Type": "application/json"
    }
  };

  const response = await fetch(url, options);
  return await response.json();
};

export const userFetch = async webToken => {
  const url = "https://pitch-battles-api.herokuapp.com/api/v1/dashboard/";

  const options = {
    method: "GET",
    mode: "cors",
    headers: {
      Authorization: `bearer ${webToken}`,
      "Content-Type": "application/json"
    }
  };

  console.log(options);

  const response = await fetch(url, options);
  return await response.json();
};

export const postGameUserUpdate = async (update, webToken) => {
  const url = "https://pitch-battles-api.herokuapp.com/api/v1/users";

  const body = update;

  const options = {
    body: JSON.stringify(body),
    method: "POST",
    mode: "cors",
    headers: {
      "Content-Type": "application/json"
    }
  };
  return;
};

export const forgotMyPasswordCall = async email => {
  // const url
  // return response
  return Promise.resolve(true);
};
