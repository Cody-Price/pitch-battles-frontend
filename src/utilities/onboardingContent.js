export const onboardingContent = [
  {
    imageUrl: "./images/battle-ob.png",
    key: "intro",
    content:
      "In Pitch Battles, musicians can test their pitch-naming mettle in combat against beasts of the wilds."
  },
  {
    imageUrl: "./images/join-class-ob.png",
    key: "join-class",
    content:
      "Before charging in, students are encouraged to join a class on the student dashboard by clicking on the 'join a class' link and entering the key given to them by their teacher."
  },
  {
    imageUrl: "./images/db-ob.png",
    key: "start-game",
    content:
      "To play Pitch Battles, choose your instrument from the menu and click 'to battle!'."
  },
  {
    imageUrl: "./images/pitch-ob.png",
    key: "play-the-game",
    content:
      "Once the game begins, you will be shown a staff with a single note. Identify it correctly by either clicking the pitch name or by entering it on the keyboard and you will strike the beast!"
  },
  {
    imageUrl: "./images/main-game-ob.png",
    key: "game-description",
    content:
      "Guess incorrectly and the beast will strike you. Each hit results in a lost heart. Knock out all the beast's hearts to advance to the next level. Lose all your hearts and it's game over..."
  },
  {
    imageUrl: "./images/db-ob.png",
    key: "badges-description",
    content:
      "Playing games, beating levels, and improving your times will earn you badges. View your badges on your dashboard or see your entire class's progress on the class page (accessed from the dashboard)"
  }
];
