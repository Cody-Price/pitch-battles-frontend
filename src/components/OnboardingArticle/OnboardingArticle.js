import React, { Component } from "react";
import PropTypes from "prop-types";

import "./OnboardingArticle.css";

class OnboardingArticle extends Component {
  render() {
    return (
      <article className="onboarding-article">
        <img
          src={this.props.data.imageUrl}
          className="onboarding-image"
          alt={this.props.data.key}
        />
        <p className="onboarding-content">{this.props.data.content}</p>
      </article>
    );
  }
}

export default OnboardingArticle;

OnboardingArticle.propTypes = {
  data: PropTypes.object
};
