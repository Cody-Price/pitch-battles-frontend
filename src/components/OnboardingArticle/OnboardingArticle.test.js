import React from "react";

import { onboardingContent } from "../../utilities/onboardingContent";
import OnboardingArticle from "./OnboardingArticle";
import Enzyme, { shallow } from "enzyme";
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("OnboardingArticle", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<OnboardingArticle data={onboardingContent[0]} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
