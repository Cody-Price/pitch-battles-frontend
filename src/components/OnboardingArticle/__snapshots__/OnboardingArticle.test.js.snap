// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingArticle should match the snapshot 1`] = `
<article
  className="onboarding-article"
>
  <img
    alt="intro"
    className="onboarding-image"
    src="./images/battle-ob.png"
  />
  <p
    className="onboarding-content"
  >
    In Pitch Battles, musicians can test their pitch-naming mettle in combat against beasts of the wilds.
  </p>
</article>
`;
